File: exercises/week05/exercise-dp-two/src/abstractfactory/ParserFactoryProducer.java
package abstractfactory;

public final class ParserFactoryProducer {

    private ParserFactoryProducer() {
        throw new AssertionError();
    }

    public static AbstractParserFactory getFactory(String factoryType) {

        switch (factoryType) {
            case "NYCFactory":
                return new NYCParserFactory();
            case "LondonFactory":
                return new LondonParserFactory();
        }

        return null;
    }

}

File: exercises/week05/exercise-dp-two-scala/src/abstractfactory/ParserFactoryProducer.scala
package abstractfactory

object ParserFactoryProducer {
  def getFactory(factoryType: String): AbstractParserFactory = {
    factoryType match {
      case "NYCFactory" =>
        new NYCParserFactory
      case "LondonFactory" =>
        new LondonParserFactory
    }
  }
}
